packages/apps/human-app/server/src/modules/statistics/statistics.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { AxiosRequestConfig } from 'axios';
+import { CommonHttpUtilService } from '../../common/utils/common-http-util.service';
+import {
+  UserStatisticsCommand,
+  UserStatisticsResponse,
+} from './interfaces/user-statistics.interface';
+import {
+  OracleStatisticsCommand,
+  OracleStatisticsResponse,
+} from './interfaces/oracle-statistics.interface';
+import { CACHE_MANAGER } from '@nestjs/cache-manager';
+import { Cache } from 'cache-manager';
+import { EnvironmentConfigService } from '../../common/config/environment-config.service';
+import { RequestContext } from '../../common/utils/request-context.util';
+
+@Injectable()
+export class StatisticsService {
+  constructor(
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
+    private httpService: CommonHttpUtilService,
+    private configService: EnvironmentConfigService,
+    private readonly requestContext: RequestContext,
+  ) {}
+  async getOracleStats(
+    command: OracleStatisticsCommand,
+  ): Promise<OracleStatisticsResponse> {
+    const url = command.oracle_url;
+    const cachedStatistics: OracleStatisticsResponse | undefined =
+      await this.cacheManager.get(url);
+    if (cachedStatistics) {
+      return cachedStatistics;
+    }
+    const options: AxiosRequestConfig = {
+      method: 'GET',
+      url: `${url}/stats`,
+    };
+    const statisticalData: OracleStatisticsResponse =
+      await this.httpService.callExternalHttpUtilRequest<OracleStatisticsResponse>(
+        options,
+      );
```

Comment:
IMO we shouldn't call an external service in the service layer, it could be behind an abstraction